### DISCUSSION TITLE
Warnings after the mda.Universe command

### DISCUSSION THREAD

**Discussion by xavgit** [MDAnalysis/mdanalysis#4384] [Q&A]
Hi,
I'm studying the quick start guide with AMBER files.
If I use `u = mda.Universe( 'target.prmtop' , '10_prod_md_combined.nc' )` the first time
I receive two warnings.
If  I use the same command the second time with the same files I don't receive any warnings.
The following is the output of the session:
```
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
\>>> import MDAnalysis as mda
\>>> u = mda.Universe( 'target.prmtop' , '10_prod_md_combined.nc' )
/home/xxxx/.local/lib/python3.10/site-packages/MDAnalysis/topology/TOPParser.py:295: UserWarning: Unknown ATOMIC_NUMBER value found for some atoms, these have been given an empty element record. If needed these can be guessed using MDAnalysis.topology.guessers.
  warnings.warn(msg)
/home/xxxx/.local/lib/python3.10/site-packages/MDAnalysis/coordinates/TRJ.py:554: UserWarning: NCDF trajectory does not contain `time` information; `time` will be set as an increasing index
  warnings.warn(wmsg)
/home/xxxx/.local/lib/python3.10/site-packages/MDAnalysis/coordinates/TRJ.py:667: UserWarning: Reader has no dt information, set to 1.0 ps
  ts.time)  # in-place ! (hope this works...)
\>>> u = mda.Universe( 'target.prmtop' , '10_prod_md_combined.nc' )
\>>> 
```
Why?
Can I safely continue with the analysis?

Thanks.

**Comment by IAlibay** [comment]
@xavgit 

**re safely proceed with analysis:** The warnings are telling you that both the `time` and `element` information in the topology file is unreliable. If your analysis relies on either of these then the answer is that no, it would not be safe to proceed without providing further information. If it doesn't rely on that information, then it would be fine.

**re why:** There are multiple different reasons for this information being missing. If I remember correctly, time information can be lost when combining restart files together, or when picking random frames with cpptraj. Element information is not guaranteed and can be missing in some older amber force field files or when using coarse grained force fields.

**re warning goes away the second time:** That's expected. Warnings from repeated operations only get raised once in a Python session otherwise you'd be constantly inundated with warnings.

**Reply by xavgit** [reply]
Ok.
Yes, for the time information warning I 've used the combination of some restart files using pytraj.iterload() and traj.save().
For the element information can I receive a reference describing it ( I'm new to the topic) so to 
understand how to provide the missing information in the case it is mandatory for the usual types
of analysis that are made with the produced trajectory.

Thanks.

Saverio

**Reply by xavgit** [reply]
Hi,
how I can detect the atoms for which holds the warnings
"Unknown ATOMIC_NUMBER value found"
as this can be important information.

Thanks.

Saverio

**Reply by IAlibay** [reply]
Atoms with missing elements will have their element atribute set to an empty string.

To find the indexes of all the atoms missing an element attribute you can do:

```python
[atom.index for atom in u.atoms if atom.element == '']
```

To find out more about how to guess atom elements please see here: https://docs.mdanalysis.org/stable/documentation_pages/topology/guessers.html#guessing-elements-from-atom-names

Please note that in some cases missing elements are expected - for example for > 3 site waters you will have missing elements for the extra particles (e.g. EPW named atoms).

**Reply by xavgit** [reply]
Hi,
thanks for the code and informations.
Yes, after stripping the :wat residues I don't receive the first UserWarning.

In the case of combined trajectory I know the values of dt, nstlim and nt* of each single trajectory. 
How do I can give this information to avoid even the time and dt warnings?

Thanks.

Saverio
